crates/polars-core/src/series/arithmetic/fixed_size_list.rs

```diff
@@ -0,0 +1,812 @@
+
+            assert!(!output_widths.is_empty());
+
+            if cfg!(debug_assertions) {
+                match (array_validities_lhs.len(), array_validities_rhs.len()) {
+                    (l, r) if l == output_widths.len() && l == r && l > 0 => {},
+                    (v, 0) | (0, v) if v == output_widths.len() => {},
+                    _ => panic!(), // One side should have been an array.
+                }
+            }
+
+            if output_len == 0
+                || (matches!(
+                    &op_apply_type,
+                    BinaryOpApplyType::ListToList | BinaryOpApplyType::ListToPrimitive
+                ) && lhs.rechunk_validity().map_or(false, |x| x.set_bits() == 0))
+                || (matches!(
+                    &op_apply_type,
+                    BinaryOpApplyType::ListToList | BinaryOpApplyType::PrimitiveToList
+                ) && rhs.rechunk_validity().map_or(false, |x| x.set_bits() == 0))
+            {
+                let DataType::Array(inner_dtype, width) = output_dtype else {
+                    unreachable!()
+                };
+
+                Ok(ArrayChunked::full_null_with_dtype(
+                    lhs.name().clone(),
+                    output_len,
+                    inner_dtype.as_ref(),
+                    width,
+                ))
+            } else {
+                Self {
+                    op,
+                    output_name: lhs.name().clone(),
+                    op_apply_type,
+                    broadcast,
+                    stride,
+                    output_widths,
+                    output_dtype,
+                    output_primitive_dtype,
+                    output_len,
+                    data_lhs: (lhs, array_validities_lhs),
+                    data_rhs: (rhs, array_validities_rhs),
+                    swapped: false,
+                }
+                .finish()
+            }
+        }
+
+        pub(super) fn finish(mut self) -> PolarsResult<ArrayChunked> {
+            // We have physical codepaths for a subset of the possible combinations of broadcasting and
+            // column types. The remaining combinations are handled by dispatching to the physical
+            // codepaths after operand swapping.
+            //
+            // # Physical impl table
+            // Legend
+            // * |  N  | // impl "N"
+            // * | [N] | // dispatches to impl "N"
+            //
+            //                  |  L  |  N  |  R  | // Broadcast (L)eft, (N)oBroadcast, (R)ight
+            // ListToList       | [1] |  0  |  1  |
+            // ListToPrimitive  |  2  |  3  |  4  |
+            // PrimitiveToList  | [4] | [3] | [2] |
+
+            self.swapped = true;
+
+            match (&self.op_apply_type, &self.broadcast) {
+                // Mostly the same as ListNumericOp, however with fixed size list we also have
+                // (BinaryOpApplyType::ListToPrimitive, Broadcast::Left) as a physical impl.
```

Comment:
Arrays have a simpler layout compared to lists, so we can also implement `array<->primitive` with the array side being broadcasted without materializing.